config/environments/production.rb

```diff
@@ -31,12 +31,15 @@
       domain: GlobalSetting.smtp_domain,
       user_name: GlobalSetting.smtp_user_name,
       password: GlobalSetting.smtp_password,
-      authentication: GlobalSetting.smtp_authentication,
       enable_starttls_auto: GlobalSetting.smtp_enable_start_tls,
       open_timeout: GlobalSetting.smtp_open_timeout,
       read_timeout: GlobalSetting.smtp_read_timeout,
     }
 
+    if settings[:password] || settings[:user_name]
+      settings[:authentication] = GlobalSetting.smtp_authentication
+    end
```

Comment:
Would it make sense to move this fix to a place where it applies for all environments? Hard to test/validate things when they are for production only. 

I was thinking of adding a similar rule here instead:  

https://github.com/discourse/discourse/blob/main/app/services/email_settings_validator.rb#L130-L132

Reply:
On second thought, there isn't another place to move this fix, this gets set in the ActionMailer configuration right below here.